spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadPropertiesAutoConfiguration.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config.reload;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnCloudPlatform;
+import org.springframework.boot.cloud.CloudPlatform;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+
+/**
+ * @author wind57
+ */
+@ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES)
+@EnableConfigurationProperties(ConfigReloadProperties.class)
+public class ConfigReloadPropertiesAutoConfiguration {
```

Comment:
I need this so that `ConfigReloadProperties` are for sure loaded before we evaluate `ConditionalOnConfigMapsReloadEnabled/ConditionalOnSecretsReloadEnabled`. As such this is used in `KubernetesClientConfigReloadAutoConfiguration` (and fabric8 equivalent) and is used in the form of:

```
@AutoConfigureAfter(ConfigReloadPropertiesAutoConfiguration.class)
public class KubernetesClientConfigReloadAutoConfiguration
```